### PR TITLE
Add `name` to `channel_create`

### PR DESCRIPTION
**Pull request by conradgrobler**
This change adds a debug name field to channels.

This is a step towards #1615.

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [x] I have written tests that cover the code changes.
  - [x] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [x] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_runtime/src/channel.rs [hunk 1]
```diff
@@ -280,6 +288,7 @@ impl Channel {
             reader_count: AtomicU64::new(0),
             waiting_threads: Mutex::new(HashMap::new()),
             label: label.clone(),
+            name: name.to_owned(),
```
Comment (juliettepretot) [inline oak_runtime/src/channel.rs:291]: For node names, we presently combine the passed name with the id. 

https://github.com/project-oak/oak/blob/ab7e7f032359e0233ef3f859a9193733a9895372/oak_runtime/src/lib.rs#L1207

Should we do the same for channels?

File: docs/abi.md [hunk 2]
```diff
@@ -181,9 +181,10 @@ If writing to the specified channel would violate
 
 ### `channel_create`
 
-Creates a new unidirectional Channel assigning the label specified by `param[2]`
-and `param[3]` to the newly created Channel, and returns the Channel handles for
-its read and write halves as output parameters in `param[0]` and `param[1]`.
+Creates a new unidirectional Channel assigning the name specified by `param[2]`
+and `param[3]` and the label specified in `param[4]` and `param[5]` to the newly
+created Channel, and returns the Channel handles for its read and write halves
+as output parameters in `param[0]` and `param[1]`.
```
Comment (tiziano88) [inline docs/abi.md:187]: Could you mention that the name may be empty, does not need to be unique, and is only used for debugging?

File: docs/programming-oak.md [hunk 3]
```diff
@@ -488,8 +488,8 @@ to the room:
 [embedmd]:# (../examples/chat/module/rust/src/lib.rs Rust /.*self\.rooms\.entry\(/ /\}\);$/)
 ```Rust
                 let channel = self.rooms.entry(label.clone()).or_insert_with(|| {
-                    let (wh, rh) =
-                        oak::io::channel_create(&label).expect("could not create channel");
+                    let (wh, rh) = oak::io::channel_create("Room initial channel", &label)
```
Comment (tiziano88) [inline docs/programming-oak.md:491]: I suspect we can just omit "channel" from any channel name, it should be pretty obvious that they are channels from the context in which they are printed / displayed (hopefully).

File: oak_runtime/src/channel.rs (hunk 1 above)
Comment (tiziano88) [reply oak_runtime/src/channel.rs:291]: I think we should too.

**Review by tiziano88 — APPROVED** [review]

File: docs/abi.md (hunk 2 above)
Comment (conradgrobler) [reply docs/abi.md:187]: Added a paragraph to explain this.

File: docs/programming-oak.md (hunk 3 above)
Comment (conradgrobler) [reply docs/programming-oak.md]: Removed "channel" from the names everywhere.

File: oak_runtime/src/channel.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_runtime/src/channel.rs]: Done.

File: sdk/rust/oak/src/grpc/mod.rs [hunk 4]
```diff
@@ -236,7 +236,7 @@ where
 {
     // Create a new channel for request message delivery.
     let (req_sender, req_receiver) =
-        crate::io::channel_create::<GrpcRequest>(&Label::public_untrusted())
+        crate::io::channel_create::<GrpcRequest>("gRPC request", &Label::public_untrusted())
```
Comment (rbehjati) [inline sdk/rust/oak/src/grpc/mod.rs:239]: What is the difference between this and 
```
 .channel_create("gRPC request", &label)
```
in `oak_runtime/src/node/grpc/server/mod.rs`? Does one of these need a different name?

**Review by rbehjati — APPROVED** [review]
Nice! :)

**Review by juliettepretot — APPROVED** [review]

File: sdk/rust/oak/src/grpc/mod.rs (hunk 4 above)
Comment (conradgrobler) [reply sdk/rust/oak/src/grpc/mod.rs:239]: I believe this is the equivalent that can used by a Wasm node to send gRPC invocations out to the gRPC client pseudo node. I think that it is fine for both to have the same name as they would be used in different contexts, which should make clear whether it is a gRPC client or server pseudo node.

**Pull request merged**